### PR TITLE
Fix issue where aggregate not updating on put

### PR DESCRIPTION
**Pull request by oxyflush**
### Problem

When a user edited an existing review, the aggregate `reviewCount` for the restaurant was being incremented as if a new review had been added. This caused the review count to drift upward over time whenever a user updated their review, corrupting the aggregate data (review count and average score) for affected restaurants.

### Root Cause

The `putReview` path was only fetching the aggregate when determining how to update it. It had no visibility into whether the account already had an existing review for the restaurant, so it always treated `putReview` as a new review submission and called `withNewReview`, unconditionally incrementing the count.

### Solution

- Introduced `getRestaurantAggregateAndExistingReview`, which uses a `TransactGetItems` call to atomically fetch both the aggregate **and** the caller's existing review in a single round trip
- `putReviewWithTransactionalAggregate` now inspects the existing review:
  - If one exists → calls `withUpdatedReview(oldScore, newScore)` on `AggregateRanking`, which adjusts `totalScore` and `averageScore` without touching `reviewCount`
  - If none exists → calls `withNewReview(newScore)` as before, incrementing `reviewCount`
- Added `AggregateRanking.withUpdatedReview` to encapsulate the score-swap arithmetic
- Strengthened the new-aggregate condition expression from `attribute_not_exists(#pk)` to `attribute_not_exists(#pk) AND attribute_not_exists(#sk)` to prevent partial aggregate creation races
- Refactored `deleteUserReview` to reuse `getRestaurantAggregateAndExistingReview`, replacing the previous two separate `GetItem` calls and reducing overall DynamoDB round trips
- Renamed `AddNewReviewForRestaurantHandler` → `PutReviewHandler` and `addNewReview` → `putReview` across the stack to accurately reflect the upsert semantics

### Tests Added

- `testPutReview_withExistingReview_doesNotIncrementReviewCount` — core regression test verifying that editing a review keeps `reviewCount` unchanged while correctly adjusting `totalScore` and `averageScore`
- `testPutReview_withExistingReview_sameScore_doesNotChangeAggregate` — edge case verifying no aggregate drift when a review is re-saved with the same score
- All existing `deleteUserReview` and `putReview` tests updated to use `TransactGetItems` mocking consistent with the new implementation

### CODE REVIEW

File: src/main/java/com/fryrank/dal/ReviewDALImpl.java [hunk 1]
```diff
@@ -300,18 +305,57 @@ public Review addNewReview(@NonNull final Review review) {
...
-        return dynamoDb.getItem(getAggregateRequest).item();
+        TransactGetItemsResponse response = dynamoDb.transactGetItems(request);
+
+        List<ItemResponse> items = response.responses();
+
+        Map<String, AttributeValue> aggregate =
+                !items.isEmpty() ? items.get(0).item() : null;
+
+        Map<String, AttributeValue> review =
+                items.size() > 1 ? items.get(1).item() : null;
```
Comment (NickPriv) [inline src/main/java/com/fryrank/dal/ReviewDALImpl.java:356]: could return empty map instead of null

File: src/main/java/com/fryrank/model/AggregateRanking.java [hunk 2]
```diff
@@ -116,4 +116,18 @@ public static AggregateRanking forFirstReview(String restaurantId, Double score)
                 .averageScore(score)
                 .build();
     }
+
+    public AggregateRanking withUpdatedReview(double oldScore, double newScore) {
+        double newTotalScore = this.totalScore - oldScore + newScore;
+        double newAverageScore = this.reviewCount == 0 ? 0.0 : newTotalScore / this.reviewCount;
```
Comment (NickPriv) [inline src/main/java/com/fryrank/model/AggregateRanking.java:122]: this is an exceptional state. can we recover from this instead of assigning a 0.0 average score?
